pennylane/_qubit_device.py

```diff
@@ -317,15 +326,18 @@ def execute(self, circuit, **kwargs):
             ):
                 # Measurements with expval or var
                 results = self._asarray(results, dtype=self.R_DTYPE)
-            elif any(ret is not qml.measurements.Counts for ret in ret_types):
-                # all the other cases except all counts
+            elif no_counts:
```

Comment:
Oh the def is `self.is_sampled = any(m.return_type is Sample for m in self.measurements)`, so it only checks `Sample`, not `Counts.

Reply:
Could we change `self.is_sampled` to any(m.return_type is Sample or Counts for m in self.measurements)? Because we actually sample to create the counts

Reply:
I Romain's suggestion that `self.is_sampled` checks for `Counts` as well. Since the measurement is constructed through `qml.sample`, that only makes sense.

Reply:
Not sure I'd agree here: `tape.is_sampled` is very specifically used when raw samples are being returned. Although samples are created for counts too, the cases in which that attribute is being used do not apply to counts. These are exactly cases in the QNode and QubitDevice, where we have dedicated cases just for Counts. Therefore, I'd be more inclined to rename the attribute itself, rather than change its definition.